### PR TITLE
Added exists and doesnt_exist method to QueryBuilder. 

### LINKED ISSUE #687: Add `.exists()` query builder method
Rather than having to pull in a model record to check for existence, or creating a custom query, it would be nice to just do `User().query().where(...).exists() == False`

### PR DESCRIPTION
**Pull request by JarriqTheTechie**
Check for the existence of rows matching your query criteria. Useful alternative to performing a row count. 

Closes #687

### CODE REVIEW

**Comment by JarriqTheTechie** [comment]
@josephmancuso ready for review.

**Comment by Marlysson** [comment]
Closes #687

File: src/masoniteorm/models/Model.py [hunk 1]
```diff
@@ -144,6 +144,8 @@ class Model(TimeStampsMixin, ObservesEvents, metaclass=ModelMeta):
         "count",
         "decrement",
         "delete",
+        "doesnt_exists",
```
Comment (josephmancuso) [inline src/masoniteorm/models/Model.py]: This method should be `doesnt_exist` i think and not `doesnt_exists`. the plural on the doesnt_exists doesn't really make gramatically sense.

Comment (josephmancuso) [reply src/masoniteorm/models/Model.py]: Nevermind this is just a typo here

**Review by josephmancuso — CHANGES_REQUESTED** [review]

**Review by josephmancuso — APPROVED** [review]

**Pull request merged**